src/engine/srv.c

```diff
@@ -454,28 +454,41 @@ dss_srv_handler(void *arg)
 		goto crt_destroy;
 	}
 
-	if (dx->dx_main_xs) {
+	if (dss_xstream_has_context(dx)) {
 		ABT_thread_attr attr;
 
 		/* Initialize NVMe context for main XS which accesses NVME */
-		rc = bio_xsctxt_alloc(&dmi->dmi_nvme_ctxt, dmi->dmi_tgt_id, false);
+		if (dx->dx_xs_id == 0) {
+			rc = bio_sys_xsctxt_alloc(&dmi->dmi_nvme_ctxt, dmi->dmi_tgt_id, false);
```

Comment:
This won't work, since SMD is required when allocating per-xstream context for sys xstream. (To store which device is assigned to sys xstream), so for sys xstream, it should do:
- bio_glbctxt_alloc();
- Initialized sysdb & setup smd;
- bio_xsctxt_alloc();

Reply:
Sure

Reply:
Above won't work, because init sysdb need sys xstream blobstore prepared, and allocating context for sys xstream don't need SMD. so right logic shall be:

1. bio_glbctxt_alloc()
    init meta/wal blobstore properly---> this don't rely on SMD.
2. init sysdb && setup smd.

Reply:
problem is if we want to use SMD in sys xs for RDB, we might need do that after step 2(SMD is ready)